doc/manual/merging.md

```diff
@@ -0,0 +1,117 @@
+# Merging records
+
+When you have a big configuration, the way to split it in several files
+(e.g.: separated by cathegorie or level of abstraction) is the merging operator.
+In nickel this is the `&` operator.
```

Comment:
We'll have to change this paragraph, it's a little awkward. And probably doesn't hammer strongly enough why the reader should read on.